### DISCUSSION TITLE
Incorrect return type when looking up for an entity that is supposed to be not null / existing

### DISCUSSION THREAD

**Discussion by LunarMelody** [prisma/prisma#17792] [Q&A]
Hello! I'm not sure if I'm doing something wrong, but for some reason this code still produces types as if `content` could be null while checking if it's not actually null :thinking: 

```typescript
const post = await prisma.post.findFirst({
  where: {
    slug,
    content: { isNot: null },
  },
  include: {
    content: true,
    author: true,
  },
});

if (!post) {
  throw plainNotFound();
}

const html = parseMarkdown(post.content.raw); // <- post.content is possibly null
```

Why is it like that? Shouldn't `post.content` type already be `Post` instead of `Post | null`?

**Comment by nurul3101** [comment]
Hello @TenkoSpirit 👋 

Is `content` relation defined as required in the post model in schema file?
Also, By default `post` type will not have relation fields included, only the scalar fields are included by default.

You would need to use Prisma.validator to create a type that includes content and author relation fields, here is a [guide](https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety/operating-against-partial-structures-of-model-types#problem-using-variations-of-the-generated-model-type) on how you can do that.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
